### PR TITLE
Implement additional xtce keywords

### LINKED ISSUE #12: XTCE parameters
Hello,

I want to use an xtce file for command and telemetry but some of its parameters are not supported by OpenC3. Would it be possible to add these parameters so that they are supported by OpenC3 please?

- Boolean
- Crc
- Absolute time parameter

Thanks,
Astrée Deshayes

### PR DESCRIPTION
**Pull request by jmthomas**
closes #12

### CODE REVIEW

**Comment by jmthomas** [comment]
@AstreeDeshayes I would love if you could try this out and see if it works. I have some TODOs in the code because i'm not sure about conversions and such.

File: openc3/spec/packets/parsers/xtce_parser_spec.rb [hunk 1]
```diff
@@ -294,6 +294,190 @@ def add_cmd_packet_item(tf, packet, item)
...
+
+        @pc.process_file(tf.path, 'TGT')
+        packet = @pc.commands['TGT']['CMD']
+        expect(packet).to_not be_nil
+        expect(packet.get_item('A').bit_offset).to eql 0
+        expect(packet.get_item('A').bit_size).to eql 16
+        expect(packet.get_item('A').data_type).to eql :BLOCK
+        # TODO: Size is 16 bits but we're doing a polynomial write which is a 64 bit float
+        # So not sure what's supposed to happen to convert a 64bit float to a 16 bit value
+        expect(packet.get_item('A').write_conversion.class).to eql PolynomialConversion
```
Comment (ryanmelt) [inline openc3/spec/packets/parsers/xtce_parser_spec.rb:473]: CRC polynomials are not polynomial conversions.  They define one of the parameters necessary for the CRC algorithm.

File: openc3/spec/packets/parsers/xtce_parser_spec.rb [hunk 2]
```diff
@@ -294,6 +294,190 @@ def add_cmd_packet_item(tf, packet, item)
...
+          file.puts "  </xtce:MetaCommand>"
+          file.puts "</xtce:MetaCommandSet>"
+        end
+
+        @pc.process_file(tf.path, 'TGT')
+        packet = @pc.commands['TGT']['CMD']
+        expect(packet).to_not be_nil
+        expect(packet.get_item('A').bit_offset).to eql 0
+        expect(packet.get_item('A').bit_size).to eql 16
+        expect(packet.get_item('A').data_type).to eql :BLOCK
```
Comment (ryanmelt) [inline openc3/spec/packets/parsers/xtce_parser_spec.rb:470]: CRCs should be UINT I think.

File: openc3/lib/openc3/packets/parsers/xtce_parser.rb [hunk 3]
```diff
@@ -178,14 +178,24 @@ def xtce_process_element(element)
       when 'ParameterTypeSet', 'EnumerationList', 'ParameterSet', 'ContainerSet',
         'EntryList', 'DefaultCalibrator', 'DefaultAlarm', 'RestrictionCriteria',
         'ComparisonList', 'MetaCommandSet', 'ArgumentTypeSet', 'ArgumentList',
-        'ArgumentAssignmentList', 'LocationInContainerInBits'
+        'ArgumentAssignmentList', 'LocationInContainerInBits', 'ReferenceTime',
+        'ErrorDetectCorrect'
         # Do Nothing
 
+      when 'CRC'
+        # TODO: Anything else here?
```
Comment (ryanmelt) [inline openc3/lib/openc3/packets/parsers/xtce_parser.rb:186]: Not sure what to do for now, probably just leave a TODO.  We could create a new CrcProcessor to check the packet in telemetry and maybe set the Crc for commands.

File: openc3/lib/openc3/packets/parsers/xtce_parser.rb [hunk 4]
```diff
@@ -289,7 +313,7 @@ def xtce_process_element(element)
         end
         return false # Already recursed
 
-      when 'PolynomialCalibrator'
+      when 'Polynomial', 'PolynomialCalibrator'
```
Comment (ryanmelt) [inline openc3/lib/openc3/packets/parsers/xtce_parser.rb:316]: CRC polynomial is not a conversion.

**Review by ryanmelt — CHANGES_REQUESTED** [review]

**Comment by AstreeDeshayes** [comment]
> @AstreeDeshayes I would love if you could try this out and see if it works. I have some TODOs in the code because i'm not sure about conversions and such.

Hello, it still doesn't work... I tried by directly putting the xtce file into the cmd_tlm folder and also with the xtce converter command line and I have the following errors : 

Processing MDB.xtce...
  Ignoring Unknown: <AbsoluteTimeParameterType>
  Ignoring Unknown: <Encoding>
  Ignoring Unknown: <ReferenceTime>
  Ignoring Unknown: <Epoch>
  Ignoring Unknown: <ErrorDetectCorrect>
  Ignoring Unknown: <CRC>
  Ignoring Unknown: <Polynomial>
  Ignoring Unknown: <Term>
  Ignoring Unknown: <Term>
  Ignoring Unknown: <Term>
  Ignoring Unknown: <Term>
  Ignoring Unknown: <BooleanParameterType>
  Ignoring Unknown: <BooleanParameterType>

**Pull request merged**